controllers/selfnoderemediation_controller.go

```diff
@@ -156,7 +165,7 @@ func (r *SelfNodeRemediationReconciler) Reconcile(ctx context.Context, req ctrl.
 		return ctrl.Result{}, r.patchSnrStatus(snr, org)
 	}
 
-	if !r.isFencingCompleted(snr) {
+	if !r.isAtRemediationPhase(snr, fencingCompletedPhase) {
```

Comment:
Nit: I think `isAtRemediationPhase` can be deleted due to latest changes.
You can use instead:
```
if r.getPhase(snr) != fencingCompletedPhase {
```